waspc/cli/Command/CreateNewProject.hs

```diff
@@ -33,7 +45,25 @@ createNewProject projectName = do
     let extCodeDir = waspProjectDir </> Common.extCodeDirInWaspProjectDir
     liftIO $ do
         createDirectorySP extCodeDir
-        writeFileSP (extCodeDir </> mainPageJsFileInExtCodeDir) mainPageJsFileContent
+        dataDir <- getAbsDataDirPath
+
+        -- Scaffold MainPage.js
+        System.Directory.copyFile
+            (SP.toFilePath (dataDir </> cliTemplatesDirInDataDir </>
```

Comment:
You should extract `dataDir </> cliTemplatesDirInDataDir`.

Reply:
You could extract whole `copyExtDirTemplate` so that it takes just two strong paths: one relative to CLI templates dir, one relative to ext dir.